src/stdlib/table.rs

```diff
@@ -56,9 +62,323 @@ pub fn load_table<'gc>(ctx: Context<'gc>) {
         }),
     );
 
+    table.set_field(ctx, "remove", Callback::from_fn(&ctx, table_remove_impl));
+
+    table.set_field(ctx, "insert", Callback::from_fn(&ctx, table_insert_impl));
+
     ctx.set_global("table", table);
 }
 
+fn prep_metaop_call<'seq, 'gc, const N: usize>(
+    ctx: Context<'gc>,
+    mut stack: Stack<'gc, '_>,
+    locals: Locals<'seq, 'gc>,
+    res: MetaResult<'gc, N>,
+) -> Option<LocalFunction<'seq>> {
+    match res {
+        MetaResult::Value(v) => {
+            stack.push_back(v);
+            None
+        }
+        MetaResult::Call(call) => {
+            stack.extend(call.args);
+            Some(locals.stash(&ctx, call.function))
+        }
+    }
+}
+
+async fn index_helper<'seq>(
+    seq: &mut AsyncSequence<'seq>,
+    table: &LocalTable<'seq>,
+    key: i64,
+    bottom: usize,
+) -> Result<(), LocalError<'seq>> {
+    let call = seq.try_enter(|ctx, locals, _, stack| {
+        let table = locals.fetch(table);
+        let call = meta_ops::index(ctx, Value::Table(table), Value::Integer(key))?;
+        Ok(prep_metaop_call(ctx, stack, locals, call))
+    })?;
+    if let Some(call) = call {
+        seq.call(&call, bottom).await?;
+        seq.enter(|_, _, _, mut stack| {
+            stack.resize(bottom + 1); // Truncate stack
```

Comment:
This is of course correct, but it makes me wonder if we shouldn't do this *within* meta_ops implementations, like always truncate to the expected number of returns for that specific metacall.

Not something that needs to change now, just thinking out loud really.

Reply:
It should be relatively simple, but right now the `Sequence` call logic skips a lot of the stack management of normal function calls; `ThreadState::return_to` currently does nothing when returning to a `Sequence`.  (When returning to a Lua frame, it truncates the stack.)

Reply:
Just to be clear, that's how it's designed right now so it's not *skipped*, it just has nothing to do (but maybe you already understood that).

We COULD make the stack protocol for callbacks more featureful and also allow them to specify a stack truncation like Lua frames, that might be simpler, but it does complicate the CallbackReturn API a little.

What I was actually suggesting was not that at all, but returning a callback that did the truncation manually. If we have a lot of places where we want to do this though, it might make sense to make it part of the stack protocol for callbacks.

Is that what you were suggesting?

Reply:
> returning a callback that did the truncation manually

If by this you mean wrapping the metacall `Function` in a Rust-implemented `Callback` that truncates the stack, it seems like a lot of unnecessary overhead?  In general, I think the async helper function is a reasonable place to put the stack management, unless we add explicit truncation to returns to `Sequence`s, which doesn't seem necessary at this point.

(That was what I was suggesting, but I also don't think it's worth the complexity at this point).

In this case, it may even have been better to pop the value off of the stack, and return it as a `LocalValue`, keeping the stack management entirely within `index_helper`.